storage/scylla_api.c

```diff
@@ -395,31 +399,138 @@ static status_t create_edge_table(CassSession* session) {
   }
+  service->session = cass_session_new();
+  service->cluster = create_cluster(host);
+  if (connect_session(service->session, service->cluster) != CASS_OK) {
+    ta_log_error("%s\n", "connect Scylla cluster fail");
+    return SC_STORAGE_CONNECT_FAIL;
+  }
+  return SC_OK;
+}
+
+status_t db_client_service_free(db_client_service_t* service) {
+  if (service == NULL) {
+    ta_log_error("Invaild input : NULL pointer\n");
+    return SC_TA_NULL;
+  }
+  if (service->session != NULL) {
+    cass_session_free(service->session);
+  }
+  if (service->cluster != NULL) {
+    cass_cluster_free(service->cluster);
+  }
+  return SC_OK;
+}
+
+status_t db_insert_transaction_into_identity_table(db_client_service_t* service, int64_t ID, const char* hash,
+                                                   db_txn_status_t status) {
+  status_t ret = SC_OK;
+  db_identity_t* identity;
+  if ((ret = db_identity_new(&identity)) != SC_OK) {
+    goto exit;
+  }
+  if ((ret = db_set_identity_hash(identity, (cass_byte_t*)hash, NUM_FLEX_TRITS_HASH)) != SC_OK) {
+    goto exit;
+  }
+
+  if ((ret = db_set_identity_ID(identity, ID)) != SC_OK) {
+    goto exit;
+  }
+
+  if ((ret = db_set_identity_status(identity, status)) != SC_OK) {
+    goto exit;
+  }
+
+  if ((ret = db_insert_identity_table(service, identity)) != SC_OK) {
+    goto exit;
+  }
+  db_identity_free(&identity);
+exit:
+  return ret;
+}
+
+status_t db_init_identity_keyspace(db_client_service_t* service, bool need_drop, const char* keyspace_name) {
   status_t ret = SC_OK;
   CassStatement* use_statement = NULL;
   char* use_query = NULL;
   /* Add contact points */
-  *cluster = create_cluster(hosts);
-  if (connect_session(session, *cluster) != CASS_OK) {
-    ta_log_error("%s\n", "connect Scylla cluster fail");
-    ret = SC_STORAGE_CONNECT_FAIL;
+
+  if ((ret = create_keyspace(service->session, keyspace_name)) != SC_OK) {
+    ta_log_error("create %s keyspace fail\n", keyspace_name);
+    goto exit;
+  }
+  ret = make_query(&use_query, "USE ", keyspace_name, "");
+  if (ret != SC_OK) {
+    ta_log_error("%s\n", "make USE keyspace query fail");
+    goto exit;
+  }
+  use_statement = cass_statement_new(use_query, 0);
+  if (execute_statement(service->session, use_statement) != CASS_OK) {
+    ta_log_error("USE keyspace %s fail\n", keyspace_name);
+    ret = SC_STORAGE_CASSANDRA_QUREY_FAIL;
     goto exit;
   }
-  if ((ret = create_permanent_keyspace(session, keyspace_name)) != SC_OK) {
+
+  if ((ret = create_identity_table(service->session, need_drop)) != SC_OK) {
+    ta_log_error("%s\n", "create identity table fail");
+    goto exit;
+  }
+
+exit:
+  free(use_query);
+  return ret;
+}
+
+status_t db_permanent_keyspace_init(db_client_service_t* service, bool need_drop, const char* keyspace_name) {
+  if (service == NULL) {
+    ta_log_error("%s\n", "Invaild input : NULL pointer");
+    return SC_TA_NULL;
+  }
+  status_t ret = SC_OK;
+  CassStatement* use_statement = NULL;
```

Comment:
Why did you call it `use_statement`?

Reply:
It indicates the CQL `USE some_keyspace_name;`